dot/core/service.go

```diff
@@ -526,47 +526,64 @@ func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
 	responseData := []*types.BlockData{}
 
 	for _, hash := range subchain {
-		data, err := s.blockState.GetBlockData(hash)
+		dataBody, err := s.blockState.GetBlockBody(hash)
 		if err != nil {
 			return err
 		}
 
 		blockData := new(types.BlockData)
 		blockData.Hash = hash
+		blockData.Body = dataBody
 
 		// TODO: checks for the existence of the following fields should be implemented once #596 is addressed.
 
 		// header
 		if blockRequest.RequestedData&1 == 1 {
-			blockData.Header = data.Header
+			retData, thisErr := s.blockState.GetHeader(hash)
+			if thisErr == nil && retData != nil {
+				blockData.Header = retData.AsOptional()
+			}
 		} else {
 			blockData.Header = optional.NewHeader(false, nil)
 		}
 
 		// body
 		if (blockRequest.RequestedData&2)>>1 == 1 {
-			blockData.Body = data.Body
+			retData, thisErr := s.blockState.GetBlockBody(hash)
+			if thisErr == nil && retData != nil {
+				blockData.Body = retData
+			}
 		} else {
 			blockData.Body = optional.NewBody(false, nil)
 		}
 
 		// receipt
 		if (blockRequest.RequestedData&4)>>2 == 1 {
-			blockData.Receipt = data.Receipt
+			blockData.Receipt, err = s.blockState.GetReceipt(hash)
+			if err != nil {
+				log.Error("[core] could not get Receipt for hash from db")
+			}
+
```

Comment:
you need to set `blockData.Receipt` as well here if err == nil, same for Justification and MessageQueue below

also, you aren't logging for Header and Body, consistency would be good

Reply:
you mean set on the db ? its already setting...
`blockData.Receipt, err = s.blockState.GetReceipt(hash)`

Reply:
updated.